Origin: commit 22c4ea424ce2e8febce04d324c5ec9898f5d534b
Author: Barton E. Schaefer <schaefer@zsh.org>
Date:   Thu Oct 9 17:54:47 2014 -0700
Description: 33403: be conservative about redirecting _call_program stderr, the caller may have already done so

diff --git a/Completion/Base/Utility/_call_program b/Completion/Base/Utility/_call_program
index b657648..010e094 100644
--- a/Completion/Base/Utility/_call_program
+++ b/Completion/Base/Utility/_call_program
@@ -2,8 +2,8 @@
 
 local tmp err_fd=-1
 
-if (( ${debug_fd:--1} > 2 ))
-then exec {err_fd}>&2	# debug_fd is saved stderr, 2 is log file
+if (( ${debug_fd:--1} > 2 )) || [[ ! -t 2 ]]
+then exec {err_fd}>&2	# debug_fd is saved stderr, 2 is trace or redirect
 else exec {err_fd}>/dev/null
 fi
 
